packages/encryption/src/lib/encryption.ts

```diff
@@ -182,49 +135,41 @@ export const decryptFromIpfs = async (
     });
 
-  try {
-    const metadata: EncryptToIpfsPayload = await (
-      await fetch(`https://gateway.pinata.cloud/ipfs/${ipfsCid}`)
-    ).json();
-    if (metadata.dataType === 'string') {
-      return decryptToString(
-        {
-          accessControlConditions: metadata.accessControlConditions,
-          evmContractConditions: metadata.evmContractConditions,
-          solRpcConditions: metadata.solRpcConditions,
-          unifiedAccessControlConditions:
-            metadata.unifiedAccessControlConditions,
-          ciphertext: metadata.ciphertext,
-          dataToEncryptHash: metadata.dataToEncryptHash,
-          chain: metadata.chain,
-          authSig,
-          sessionSigs,
-        },
-        litNodeClient
-      );
-    } else {
-      return decryptToFile(
-        {
-          accessControlConditions: metadata.accessControlConditions,
-          evmContractConditions: metadata.evmContractConditions,
-          solRpcConditions: metadata.solRpcConditions,
-          unifiedAccessControlConditions:
-            metadata.unifiedAccessControlConditions,
-          ciphertext: metadata.ciphertext,
-          dataToEncryptHash: metadata.dataToEncryptHash,
-          chain: metadata.chain,
-          authSig,
-          sessionSigs,
-        },
-        litNodeClient
-      );
-    }
-  } catch (e) {
-    return throwError({
-      message: 'Unable to fetch or decrypt from IPFS',
-      errorKind: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.kind,
-      errorCode: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.name,
-    });
+  // FIXME: The return type of this function is inferrable based on the value of `params.dataType`
+  if (parsedJsonData.dataType === 'string') {
+    return decryptToString(
+      {
+        accessControlConditions: parsedJsonData.accessControlConditions,
+        evmContractConditions: parsedJsonData.evmContractConditions,
+        solRpcConditions: parsedJsonData.solRpcConditions,
+        unifiedAccessControlConditions:
+          parsedJsonData.unifiedAccessControlConditions,
+        ciphertext: parsedJsonData.ciphertext,
+        dataToEncryptHash: parsedJsonData.dataToEncryptHash,
+        chain: parsedJsonData.chain,
+        authSig,
+        sessionSigs,
+      },
+      litNodeClient
+    );
+  } else if (parsedJsonData.dataType === 'file') {
+    return decryptToFile(
+      {
+        accessControlConditions: parsedJsonData.accessControlConditions,
+        evmContractConditions: parsedJsonData.evmContractConditions,
+        solRpcConditions: parsedJsonData.solRpcConditions,
+        unifiedAccessControlConditions:
+          parsedJsonData.unifiedAccessControlConditions,
+        ciphertext: parsedJsonData.ciphertext,
+        dataToEncryptHash: parsedJsonData.dataToEncryptHash,
+        chain: parsedJsonData.chain,
+        authSig,
+        sessionSigs,
+      },
+      litNodeClient
+    );
+  } else {
+    throw new Error(`Datatype "${parsedJsonData.dataType}" is not supported.`);
```

Comment:
Can we also log the supported types out?

Reply:
:+1: Updated this to return the same string as our `paramsValidator` does, which mentions the possible strings